core/src/test/scala/com/avast/cloud/metrics/datadog/impl/TimerImplTest.scala

```diff
@@ -0,0 +1,44 @@
+package com.avast.cloud.metrics.datadog.impl
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect.{ Clock, IO }
+import com.avast.cloud.metrics.datadog.api.Tag
+import com.timgroup.statsd.StatsDClient
+import org.mockito.scalatest.MockitoSugar
+import org.scalatest.{ Assertions, BeforeAndAfter, FlatSpec }
+
+class TimerImplTest extends FlatSpec with MockitoSugar with BeforeAndAfter with Assertions {
+
+  trait Fixtures {
+    val aspect: String = "metric"
+    val sampleRate     = 1
+
+    val statsD: StatsDClient = mock[StatsDClient]
```

Comment:
isn't there better way how to mock `StatsDClient`? I don't really like mocking frameworks as it makes tests very fragile.

Reply:
i don't like it either, but statsdclient has like 80 methods that would need to be overwritten - i don't think it's worth it for case this simple